internal/kubernetes/connection.go

```diff
@@ -18,12 +22,18 @@ func CheckConnection(ctx context.Context, informer validation.Informer, node *ap
 		informer.Done(ctx, name, err)
 	}()
 
-	endpoint, err := url.Parse(node.Spec.Cluster.APIServerEndpoint)
+	endpoint, err := url.ParseRequestURI(node.Spec.Cluster.APIServerEndpoint)
```

Comment:
What is the reason behind this change?

Reply:
ParseRequestURI parses the endpoint so that we can directly use the hostname to check if DNS resolution. This change should be fine since the APIServer endpoint is a HTTP endpoint